tools/make_release_tarball

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+if [ ! -e tools/make_release_tarball  -o  "$1" != "" ]
+then
+  echo "Handy script for creating tarball: Use from the checkout directory"
+  exit 0
+fi
+
+set -e
+
+VERSION_MAJOR=$(grep set.MIR_VERSION_MAJOR CMakeLists.txt | cut -f2 -d' ' | tr -d \))
+VERSION_MINOR=$(grep set.MIR_VERSION_MINOR CMakeLists.txt | cut -f2 -d' ' | tr -d \))
+VERSION_PATCH=$(grep set.MIR_VERSION_PATCH CMakeLists.txt | cut -f2 -d' ' | tr -d \))
+
+VERSIONED_NAME=mir-$VERSION_MAJOR.$VERSION_MINOR.$VERSION_PATCH
+
+SCRATCH_DIR=$(mktemp -d)
+BUILD_DIR=$(mktemp -d)
+INSTALL_DIR=$(mktemp -d)
+
+function cleanup() {
+	ARG=$?
+	[ -d $SCRATCH_DIR ] && rm -r $SCRATCH_DIR
+	[ -d $BUILD_DIR ] && rm -r $BUILD_DIR
+	[ -d $INSTALL_DIR ] && rm -r $INSTALL_DIR
+	exit $ARG
+}
+
+trap cleanup EXIT
+
+echo "Generating Mir tarball…"
+pushd tests/acceptance-tests/wayland/wlcs
+git archive --format=tar -o $SCRATCH_DIR/$VERSIONED_NAME-wlcs.tar --prefix=$VERSIONED_NAME/tests/acceptance-tests/wayland/wlcs/ HEAD
+popd
+git archive --format=tar -o $SCRATCH_DIR/$VERSIONED_NAME.tar --prefix=$VERSIONED_NAME/ HEAD
+tar --concatenate -vf $SCRATCH_DIR/$VERSIONED_NAME.tar $SCRATCH_DIR/$VERSIONED_NAME-wlcs.tar
+xz -9 $SCRATCH_DIR/$VERSIONED_NAME.tar
+
+(cd $SCRATCH_DIR; tar xvJf $SCRATCH_DIR/$VERSIONED_NAME.tar.xz)
+
+echo "Testing that the tarball is buildable"
+(cd $BUILD_DIR ; cmake $SCRATCH_DIR/$VERSIONED_NAME )
+make -C $BUILD_DIR -j $(nproc)
+
+echo "Testing that the tarball is installable"
+make -C $BUILD_DIR install DESTDIR=$INSTALL_DIR
+
+mv $SCRATCH_DIR/$VERSIONED_NAME.tar.xz .
+echo "$VERSIONED_NAME.tar.xz successfully created and tested"
```

Comment:
Could everything after the release tarball be optional? Or a separate script?

Reply:
It *could* be, but I expect we're only going to be running this script a handful of times a month. Why would you want to run it without testing that the tarball is actually buildable and installable?

Reply:
Do I could use it to build tarballs to upload to the staging PPA ;)

Reply:
I've added an (undocumented) "--skip-checks" option